In [2]:
from google.colab import drive

CONTENT_DRIVE_PATH = '/content/drive'
drive.mount(CONTENT_DRIVE_PATH, force_remount=True)

Mounted at /content/drive


In [3]:
from pathlib import Path
from collections import defaultdict
import librosa

WAV_FILES_FOLDER_PATH = '/content/drive/MyDrive/MGCProject/wav_audio_files'

**this example is for the FMA medium dataset**

------------------------------------

getting the sampling rate counts:

In [ ]:
def load_audio_file(wav_file_path):
    audio_signal, sampling_rate = librosa.load(wav_file_path, sr=None)
    return audio_signal, sampling_rate

def print_sr_counts(folder_path):
    folder_path = Path(folder_path)

    sr_count_dict = defaultdict(int)

    for wav_file in folder_path.rglob('*.wav'):
        audio_signal, sampling_rate = load_audio_file(wav_file)
        sr_count_dict[sampling_rate] += 1

    print(sr_count_dict)

print_length_and_sr_counts(WAV_FILES_FOLDER_PATH)

the most common sampling rate is 44100
___________________________

getting the length counts:

In [4]:
def load_audio_file(wav_file_path):
    audio_signal, sampling_rate = librosa.load(wav_file_path, sr=44100)
    return audio_signal, sampling_rate

def get_length_counts(folder_path):
    folder_path = Path(folder_path)

    length_count_dict = defaultdict(int)
    length_ids_dict = defaultdict(list)

    for wav_file in folder_path.rglob('*.wav'):
        track_id = int(wav_file.stem)
        audio_signal, sampling_rate = load_audio_file(wav_file)
        length_count_dict[len(audio_signal)] += 1
        length_ids_dict[len(audio_signal)].append(track_id)

    return length_count_dict, length_ids_dict

length_count_dict, length_ids_dict = get_length_counts(WAV_FILES_FOLDER_PATH)

print(length_count_dict)

defaultdict(<class 'int'>, {1321967: 3344, 1323119: 2623, 1322514: 280, 1322496: 1, 22511: 1, 67439: 1, 1323330: 6, 1321742: 1, 1321438: 12, 1322590: 12})


length counts:

{1321967: 3344, 1323119: 2623, 1322514: 280, 1322496: 1, 22511: 1, 67439: 1, 1323330: 6, 1321742: 1, 1321438: 12, 1322590: 12}

{44100: 7997}

samples of length 67439, 22511 will be removed from the dataset

these are samples with ids 98569 and 98567

In [5]:
print(length_ids_dict[67439])
print(length_ids_dict[22511])

[98569]
[98567]


for an audio sampled at 44100 Hz to be 30 s long, it needs to have:

30 * 44 100 = 1 323 000 samples

so the clips longer than that will be truncated and shorter than that will be zero padded